cmd/data.go

```diff
@@ -114,6 +116,14 @@ func (cmd *DataCmd) Execute(ctx context.Context, f *flag.FlagSet, _ ...interface
 		err = fmt.Errorf("error while preparing prerequisites for migration: %v", err)
 		return subcommands.ExitUsageError
 	}
+	if cmd.project == "" {
+		getInfo := &utils.GetUtilInfoImpl{}
+		cmd.project, err = getInfo.GetProject()
+		if err != nil {
+			logger.Log.Error("Could not get project id from gcloud environment or --project flag. Inferring the migration project id from target profile.", zap.Error(err))
+			cmd.project = targetProfile.Conn.Sp.Project
```

Comment:
Why are we doing this? For backward compatibility?

Reply:
If we are proceeding ahead it should not be an ERROR log, maybe WARN. 

I think we should just fail though and not use the targetProfile value as a fallback

Reply:
Done.